src/filepattern/cpp/util/vector_parser.cpp

```diff
@@ -1,31 +1,124 @@
 #include "vector_parser.hpp"
+#include "../internal/vectorpattern.hpp"
 
 using namespace std; 
 
 string VectorParser::getFileName(string& stitching_vector_line){
-    string temp = stitching_vector_line;
-    s::removeSpaces(temp); 
-    return s::split(s::split(temp, ";")[0], ":")[1]; // return only the filename from the line
+
+    std::regex pattern =std::regex("file:\\s*(.*?);");
+
+    smatch sm;
+
+    if (regex_search(stitching_vector_line, sm, pattern)){
+        return sm[1];
+    } else {
+        throw std::runtime_error("Filename not found in the line: " + stitching_vector_line);
+    }
 }
 
+bool VectorParser::isStitchingVector(std::string& line) {
+
+    try {
+
+        Tuple temp;
+        std::vector<std::string> vars;
+
+        // attempt to parse lin
+        parseVectorLine(temp, 
+                        line, 
+                        VectorPattern::STITCH_VARIABLES_, 
+                        VectorPattern::STITCH_REGEX_VECTOR_,
+                        vars);
+
+        // no exception thrown -> valid stitching vector
+        return true;               
+    } catch (std::exception& e){
+
+        // line did not have format of stitching vector 
+        return false;
+    }
+};
+
 void VectorParser::parseVectorLine(Tuple& tup,
-                                    string& stitching_vector_line, 
-                                    const vector<string>& STITCH_VARIABLES, 
-                                    const regex& STITCH_REGEX,
-                                    vector<string> variables) {
+                                    std::string& stitching_vector_line, 
```

Comment:
use `const std::string` ?

Reply:
Curious why you did not think for returning a `Tuple` instead of passing a non-const reference and modifying it?